contracts/poolMaster/PoolMaster.sol

```diff
@@ -0,0 +1,336 @@
+   */
+  function vote(uint256[] calldata proposalIds, uint256[] calldata optionBitMasks)
+    external
+    onlyOperator
+  {
+    require(proposalIds.length == optionBitMasks.length, 'invalid length');
+    for (uint256 i = 0; i < proposalIds.length; i++) {
+      kyberGovernance.submitVote(proposalIds[i], optionBitMasks[i]);
+    }
+  }
+
+  /*
+   * @notice Claim accumulated reward thus far
+   * @notice Will apply admin fee to KNC token.
+   * Admin fee for other tokens applied after liquidation to KNC
+   * @dev Admin or operator calls with relevant params
+   * @param cycle - sourced from Kyber API
+   * @param index - sourced from Kyber API
+   * @param tokens - ERC20 fee tokens
+   * @param merkleProof - sourced from Kyber API
+   */
+  function claimReward(
+    uint256 cycle,
+    uint256 index,
+    IERC20Ext[] calldata tokens,
+    uint256[] calldata cumulativeAmounts,
+    bytes32[] calldata merkleProof
+  ) external onlyOperator {
+    rewardsDistributor.claim(cycle, index, address(this), tokens, cumulativeAmounts, merkleProof);
+
+    for (uint256 i = 0; i < tokens.length; i++) {
+      if (tokens[i] == newKnc) {
+        uint256 availableKnc = _administerAdminFee(
+          FeeTypes.CLAIM,
+          getAvailableNewKncBalanceTwei()
+        );
+        _stake(availableKnc);
+      }
+    }
+  }
+
+  /*
+   * @notice Will liquidate ETH or ERC20 tokens to KNC
+   * @notice Will apply admin fee after liquidations
+   * @dev Admin or operator calls with relevant params
+   * @param tokens - ETH / ERC20 tokens to be liquidated to KNC
+   * @param minRates - kyberProxy.getExpectedRate(eth/token => knc)
+   */
+  function liquidateTokensToKnc(IERC20Ext[] calldata tokens, uint256[] calldata minRates)
+    external
+    onlyOperator
+  {
+    require(tokens.length == minRates.length, 'unequal lengths');
+    for (uint256 i = 0; i < tokens.length; i++) {
+      if (tokens[i] == ETH_ADDRESS) {
+        kyberProxy.swapEtherToToken{value: address(this).balance}(newKnc, minRates[i]);
+      } else if (tokens[i] != newKnc) {
+        kyberProxy.swapTokenToToken(
+          tokens[i],
+          tokens[i].balanceOf(address(this)),
+          newKnc,
+          minRates[i]
+        );
+      }
+    }
+    uint256 availableKnc = _administerAdminFee(FeeTypes.CLAIM, getAvailableNewKncBalanceTwei());
+    _stake(availableKnc);
+  }
+
+  /*
+   * @notice Called by admin on deployment for KNC
+   * @dev Approves Kyber Proxy contract to trade KNC
+   * @param Token to approve on proxy contract
+   * @param Pass _giveAllowance as true to give max allowance, otherwise resets to zero
+   */
+  function approveKyberProxyContract(IERC20Ext token, bool giveAllowance) external onlyOperator {
+    require(token != newKnc, 'knc not allowed');
+    uint256 amount = giveAllowance ? type(uint256).max : 0;
+    token.safeApprove(address(kyberProxy), amount);
+  }
+
+  function withdrawAdminFee() external onlyOperator {
```

Comment:
also some comments for these 2 functions, may be better to leave 1 twei left for both

Reply:
yeah, save gas. thanks